### ISSUE TITLE
Check for reuse potential 

### ISSUE THREAD

**Issue by kmadathil** [kmadathil/sanskrit_parser#4]
--- From Avinash Varna

Some of these sub routines (such as sandhi splitting and lexical validation routines mentioned in Level 1) are available in Dr. Dhaval's projects (e.g. https://github.com/drdhaval2785/inriaxmlwrapper) and Michael Bykov's projects (https://github.com/mbykov). Could leverage them to bootstrap the setup.

**Comment by kmadathil** [comment]
https://github.com/kmadathil/inriaxmlwrapper is reusable for Lexical validation with minor modifications.

**Comment by drdhaval2785** [comment]
Great. You found it somewhat useful.

**Comment by kmadathil** [comment]
More than somewhat - it does almost all of the job!
I considered reusing the sandhi split, but it runs too slow to be useful. I'll reimplement it from scratch.

**Comment by drdhaval2785** [comment]
The best way to move forward in sandhi split is the answer of Generic Human in this stackoverflow thread.
https://stackoverflow.com/questions/8870261/how-to-split-text-without-spaces-into-list-of-words

It is extremely fast.

This gives very good results for infering spaces in unspaced text.

**Comment by drdhaval2785** [comment]
https://github.com/drdhaval2785/samasasplitter 
Specifically split.py is the code which you may like to tweak to suit your needs.

There are two type of usages
1. Individual usage
`python split.py aDigrahaRa MD`
2. batch usage
`python split.py batchprocess/input.txt MW batchprocess/output.txt`.

Time frame is reasonably speedy.

The issue where this progress got stuck was with larger headwords, the permutation combinations get wilder.
It was just a question of limiting the permutation and combinations based on the previous string.
This is the problem which I and @mbykov tried to solve, but didn't succeed much.

**Comment by drdhaval2785** [comment]
Below is an example with timestamps. 
Total 0.26 seconds spent for calculation of dictionary headword costs and maxword (non-repetitive work).
0.05 seconds were spent for finding valid splits of the word.
This is reasonable speed for most of the small to medium words.
See the log below.
Word tested was prakalpanEkaSilpin.
```
$ python split.py prakalpanEkaSilpin PW
Reading knownpairs 2017-07-02 19:50:57.014000
Calculating costs of dictionary headwords 2017-07-02 19:50:57.169000
Calculated costs of dictionary headwords 2017-07-02 19:50:57.265000
Calculated maxword 2017-07-02 19:50:57.279000
valid permutations are 3
2017-07-02 19:50:57.317000
['prakalpana+eka+Silpin', 'prakalpana+Eka+Silpin', 'pra+kalpa+nEka+Silpin'] 5
2017-07-02 19:50:57.320000
```

**Comment by kmadathil** [comment]
I looked at this, @drdhaval2785 , but I'm uncomfortable with a Zipf's law approach at this stage. I'd like to first get all possible Sandhi splits, and then decide amongst them based on morphological constraints rather than word probability. A long word might be improbable, but other constraints in the sentence might make it the most probable. Thoughts? 

Also, I'm not sure if I've read your code right, but you seem to be building a set of strings with the outer product of all possible Sandhi replacement combinations for your input string, and then splitting each of them. If I read that right, this is sub-optimal for performance. I'd do this instead:

dynamic_scoreboard={}
def possible_splits(s):
  splits = []
  # Dynamic programming - remember substrings that've been seen before
  if s in dynamic_scoreboard:
     return dynamic_scoreboard[s]
  for c in s:
     do all possible sandhi replacements of c, get tlist= [(s_c_left0, s_c_right0), ...]
     for (t_left,t_right) in s_c_list:
         if is_valid_word(s_c_left):
              ps = possible_splits(s_c_right)
              if ps is not None:
                   splits.extend([s_c_left,ps])
   return splits    

Then I'd flatten the result at one go.

**Comment by avinashvarna** [comment]
Agree with @kmadathil on the dynamic programming. I've been playing with the find_words subroutine in the inriaxmlwrapper/sanskritmark.py which appears to be based on unutbu's answer in the same stackoverflow thread and is closer to the recursive function that @kmadathil outlined above. Adding the dynamic scoreboard/memoization technique + parsing all xmls and prebuilding a dict of all words appears to halve the amount of time required for analyzing the words. Haven't looked at split.py in https://github.com/drdhaval2785/samasasplitter yet. @drdhaval2785 - is that the latest?

**Comment by kmadathil** [comment]
This is the result of the pseudocode above + an optimized quick version of lookup in sanskritmark.py. For sandhi split, we don't need to know the actual tags themselves, so I created a cache of all forms as a python set, and just check for membership. Post split, we can tag each split member once.

Now really fast!

$ python SanskritLexicalAnalyzer.py astyuttarasyAMdishi --split
Parsing of XMLs started at 2017-07-05 20:37:34.052617
666994 forms cached for quick search
Parsing of XMLs completed at 2017-07-05 20:37:39.022183
astyuttarasyAMdishi
astyuttarasyAMdiSi
Start split: 2017-07-05 20:37:41.071804
End split: 2017-07-05 20:37:41.072358
[[u'asti', u'ut', u'tara', u'syAm', u'diSi'], [u'asti', u'ut', u'taras', u'yAm', u'diSi'], [u'asti', u'uttas', u'asyAm', u'diSi'], [u'asti', u'uttara', u'syAm', u'diSi'], [u'asti', u'uttaras', u'yAm', u'diSi'], [u'asti', u'uttarasyAm', u'diSi']]

I tried a longer split. We're still really fast, but now limited by sandhi quality. I will open an issue on improving sandhi using left/right contexts separately. Modulo this, we have the core of a solution all the way to Level 1. 

$ python SanskritLexicalAnalyzer.py astyuttarasyAmdishidevatAtmA --split
Parsing of XMLs started at 2017-07-05 20:34:31.200062
666994 forms cached for quick search
Parsing of XMLs completed at 2017-07-05 20:34:36.124336
astyuttarasyAmdishidevatAtmA
astyuttarasyAmdiSidevatAtmA
Start split: 2017-07-05 20:34:38.136929
End split: 2017-07-05 20:34:38.139460
[[u'asti', u'ut', u'tara', u'syAm', u'diSi', u'de', u'vatA', u'at', u'mA'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'de', u'vatA', u'at', u'mAs'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'de', u'vatA', u'A', u'at', u'mA'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'de', u'vatA', u'A', u'at', u'mAs'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'de', u'vatA', u'AtmA'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'de', u'vatA', u'Atma'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'de', u'vatA', u'AtmAs'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'de', u'vatAt', u'mA'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'de', u'vatAt', u'mAs'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'de', u'avat', u'A', u'at', u'mA'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'de', u'avat', u'A', u'at', u'mAs'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'de', u'avat', u'AtmA'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'de', u'avat', u'Atma'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'de', u'avat', u'AtmAs'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'de', u'avata', u'at', u'mA'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'de', u'avata', u'at', u'mAs'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'de', u'avata', u'A', u'at', u'mA'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'de', u'avata', u'A', u'at', u'mAs'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'de', u'avata', u'AtmA'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'de', u'avata', u'Atma'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'de', u'avata', u'AtmAs'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'de', u'avatA', u'at', u'mA'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'de', u'avatA', u'at', u'mAs'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'de', u'avatA', u'A', u'at', u'mA'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'de', u'avatA', u'A', u'at', u'mAs'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'de', u'avatA', u'AtmA'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'de', u'avatA', u'Atma'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'de', u'avatA', u'AtmAs'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'devat', u'A', u'at', u'mA'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'devat', u'A', u'at', u'mAs'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'devat', u'AtmA'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'devat', u'Atma'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'devat', u'AtmAs'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'devata', u'at', u'mA'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'devata', u'at', u'mAs'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'devata', u'A', u'at', u'mA'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'devata', u'A', u'at', u'mAs'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'devata', u'AtmA'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'devata', u'Atma'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'devata', u'AtmAs'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'devatA', u'at', u'mA'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'devatA', u'at', u'mAs'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'devatA', u'A', u'at', u'mA'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'devatA', u'A', u'at', u'mAs'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'devatA', u'AtmA'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'devatA', u'Atma'], [u'asti', u'ut', u'tara', u'syAm', u'diSi', u'devatA', u'AtmAs'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'de', u'vatA', u'at', u'mA'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'de', u'vatA', u'at', u'mAs'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'de', u'vatA', u'A', u'at', u'mA'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'de', u'vatA', u'A', u'at', u'mAs'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'de', u'vatA', u'AtmA'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'de', u'vatA', u'Atma'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'de', u'vatA', u'AtmAs'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'de', u'vatAt', u'mA'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'de', u'vatAt', u'mAs'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'de', u'avat', u'A', u'at', u'mA'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'de', u'avat', u'A', u'at', u'mAs'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'de', u'avat', u'AtmA'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'de', u'avat', u'Atma'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'de', u'avat', u'AtmAs'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'de', u'avata', u'at', u'mA'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'de', u'avata', u'at', u'mAs'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'de', u'avata', u'A', u'at', u'mA'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'de', u'avata', u'A', u'at', u'mAs'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'de', u'avata', u'AtmA'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'de', u'avata', u'Atma'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'de', u'avata', u'AtmAs'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'de', u'avatA', u'at', u'mA'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'de', u'avatA', u'at', u'mAs'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'de', u'avatA', u'A', u'at', u'mA'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'de', u'avatA', u'A', u'at', u'mAs'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'de', u'avatA', u'AtmA'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'de', u'avatA', u'Atma'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'de', u'avatA', u'AtmAs'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'devat', u'A', u'at', u'mA'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'devat', u'A', u'at', u'mAs'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'devat', u'AtmA'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'devat', u'Atma'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'devat', u'AtmAs'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'devata', u'at', u'mA'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'devata', u'at', u'mAs'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'devata', u'A', u'at', u'mA'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'devata', u'A', u'at', u'mAs'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'devata', u'AtmA'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'devata', u'Atma'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'devata', u'AtmAs'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'devatA', u'at', u'mA'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'devatA', u'at', u'mAs'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'devatA', u'A', u'at', u'mA'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'devatA', u'A', u'at', u'mAs'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'devatA', u'AtmA'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'devatA', u'Atma'], [u'asti', u'ut', u'taras', u'yAm', u'diSi', u'devatA', u'AtmAs'], [u'asti', u'uttas', u'asyAm', u'diSi', u'de', u'vatA', u'at', u'mA'], [u'asti', u'uttas', u'asyAm', u'diSi', u'de', u'vatA', u'at', u'mAs'], [u'asti', u'uttas', u'asyAm', u'diSi', u'de', u'vatA', u'A', u'at', u'mA'], [u'asti', u'uttas', u'asyAm', u'diSi', u'de', u'vatA', u'A', u'at', u'mAs'], [u'asti', u'uttas', u'asyAm', u'diSi', u'de', u'vatA', u'AtmA'], [u'asti', u'uttas', u'asyAm', u'diSi', u'de', u'vatA', u'Atma'], [u'asti', u'uttas', u'asyAm', u'diSi', u'de', u'vatA', u'AtmAs'], [u'asti', u'uttas', u'asyAm', u'diSi', u'de', u'vatAt', u'mA'], [u'asti', u'uttas', u'asyAm', u'diSi', u'de', u'vatAt', u'mAs'], [u'asti', u'uttas', u'asyAm', u'diSi', u'de', u'avat', u'A', u'at', u'mA'], [u'asti', u'uttas', u'asyAm', u'diSi', u'de', u'avat', u'A', u'at', u'mAs'], [u'asti', u'uttas', u'asyAm', u'diSi', u'de', u'avat', u'AtmA'], [u'asti', u'uttas', u'asyAm', u'diSi', u'de', u'avat', u'Atma'], [u'asti', u'uttas', u'asyAm', u'diSi', u'de', u'avat', u'AtmAs'], [u'asti', u'uttas', u'asyAm', u'diSi', u'de', u'avata', u'at', u'mA'], [u'asti', u'uttas', u'asyAm', u'diSi', u'de', u'avata', u'at', u'mAs'], [u'asti', u'uttas', u'asyAm', u'diSi', u'de', u'avata', u'A', u'at', u'mA'], [u'asti', u'uttas', u'asyAm', u'diSi', u'de', u'avata', u'A', u'at', u'mAs'], [u'asti', u'uttas', u'asyAm', u'diSi', u'de', u'avata', u'AtmA'], [u'asti', u'uttas', u'asyAm', u'diSi', u'de', u'avata', u'Atma'], [u'asti', u'uttas', u'asyAm', u'diSi', u'de', u'avata', u'AtmAs'], [u'asti', u'uttas', u'asyAm', u'diSi', u'de', u'avatA', u'at', u'mA'], [u'asti', u'uttas', u'asyAm', u'diSi', u'de', u'avatA', u'at', u'mAs'], [u'asti', u'uttas', u'asyAm', u'diSi', u'de', u'avatA', u'A', u'at', u'mA'], [u'asti', u'uttas', u'asyAm', u'diSi', u'de', u'avatA', u'A', u'at', u'mAs'], [u'asti', u'uttas', u'asyAm', u'diSi', u'de', u'avatA', u'AtmA'], [u'asti', u'uttas', u'asyAm', u'diSi', u'de', u'avatA', u'Atma'], [u'asti', u'uttas', u'asyAm', u'diSi', u'de', u'avatA', u'AtmAs'], [u'asti', u'uttas', u'asyAm', u'diSi', u'devat', u'A', u'at', u'mA'], [u'asti', u'uttas', u'asyAm', u'diSi', u'devat', u'A', u'at', u'mAs'], [u'asti', u'uttas', u'asyAm', u'diSi', u'devat', u'AtmA'], [u'asti', u'uttas', u'asyAm', u'diSi', u'devat', u'Atma'], [u'asti', u'uttas', u'asyAm', u'diSi', u'devat', u'AtmAs'], [u'asti', u'uttas', u'asyAm', u'diSi', u'devata', u'at', u'mA'], [u'asti', u'uttas', u'asyAm', u'diSi', u'devata', u'at', u'mAs'], [u'asti', u'uttas', u'asyAm', u'diSi', u'devata', u'A', u'at', u'mA'], [u'asti', u'uttas', u'asyAm', u'diSi', u'devata', u'A', u'at', u'mAs'], [u'asti', u'uttas', u'asyAm', u'diSi', u'devata', u'AtmA'], [u'asti', u'uttas', u'asyAm', u'diSi', u'devata', u'Atma'], [u'asti', u'uttas', u'asyAm', u'diSi', u'devata', u'AtmAs'], [u'asti', u'uttas', u'asyAm', u'diSi', u'devatA', u'at', u'mA'], [u'asti', u'uttas', u'asyAm', u'diSi', u'devatA', u'at', u'mAs'], [u'asti', u'uttas', u'asyAm', u'diSi', u'devatA', u'A', u'at', u'mA'], [u'asti', u'uttas', u'asyAm', u'diSi', u'devatA', u'A', u'at', u'mAs'], [u'asti', u'uttas', u'asyAm', u'diSi', u'devatA', u'AtmA'], [u'asti', u'uttas', u'asyAm', u'diSi', u'devatA', u'Atma'], [u'asti', u'uttas', u'asyAm', u'diSi', u'devatA', u'AtmAs'], [u'asti', u'uttara', u'syAm', u'diSi', u'de', u'vatA', u'at', u'mA'], [u'asti', u'uttara', u'syAm', u'diSi', u'de', u'vatA', u'at', u'mAs'], [u'asti', u'uttara', u'syAm', u'diSi', u'de', u'vatA', u'A', u'at', u'mA'], [u'asti', u'uttara', u'syAm', u'diSi', u'de', u'vatA', u'A', u'at', u'mAs'], [u'asti', u'uttara', u'syAm', u'diSi', u'de', u'vatA', u'AtmA'], [u'asti', u'uttara', u'syAm', u'diSi', u'de', u'vatA', u'Atma'], [u'asti', u'uttara', u'syAm', u'diSi', u'de', u'vatA', u'AtmAs'], [u'asti', u'uttara', u'syAm', u'diSi', u'de', u'vatAt', u'mA'], [u'asti', u'uttara', u'syAm', u'diSi', u'de', u'vatAt', u'mAs'], [u'asti', u'uttara', u'syAm', u'diSi', u'de', u'avat', u'A', u'at', u'mA'], [u'asti', u'uttara', u'syAm', u'diSi', u'de', u'avat', u'A', u'at', u'mAs'], [u'asti', u'uttara', u'syAm', u'diSi', u'de', u'avat', u'AtmA'], [u'asti', u'uttara', u'syAm', u'diSi', u'de', u'avat', u'Atma'], [u'asti', u'uttara', u'syAm', u'diSi', u'de', u'avat', u'AtmAs'], [u'asti', u'uttara', u'syAm', u'diSi', u'de', u'avata', u'at', u'mA'], [u'asti', u'uttara', u'syAm', u'diSi', u'de', u'avata', u'at', u'mAs'], [u'asti', u'uttara', u'syAm', u'diSi', u'de', u'avata', u'A', u'at', u'mA'], [u'asti', u'uttara', u'syAm', u'diSi', u'de', u'avata', u'A', u'at', u'mAs'], [u'asti', u'uttara', u'syAm', u'diSi', u'de', u'avata', u'AtmA'], [u'asti', u'uttara', u'syAm', u'diSi', u'de', u'avata', u'Atma'], [u'asti', u'uttara', u'syAm', u'diSi', u'de', u'avata', u'AtmAs'], [u'asti', u'uttara', u'syAm', u'diSi', u'de', u'avatA', u'at', u'mA'], [u'asti', u'uttara', u'syAm', u'diSi', u'de', u'avatA', u'at', u'mAs'], [u'asti', u'uttara', u'syAm', u'diSi', u'de', u'avatA', u'A', u'at', u'mA'], [u'asti', u'uttara', u'syAm', u'diSi', u'de', u'avatA', u'A', u'at', u'mAs'], [u'asti', u'uttara', u'syAm', u'diSi', u'de', u'avatA', u'AtmA'], [u'asti', u'uttara', u'syAm', u'diSi', u'de', u'avatA', u'Atma'], [u'asti', u'uttara', u'syAm', u'diSi', u'de', u'avatA', u'AtmAs'], [u'asti', u'uttara', u'syAm', u'diSi', u'devat', u'A', u'at', u'mA'], [u'asti', u'uttara', u'syAm', u'diSi', u'devat', u'A', u'at', u'mAs'], [u'asti', u'uttara', u'syAm', u'diSi', u'devat', u'AtmA'], [u'asti', u'uttara', u'syAm', u'diSi', u'devat', u'Atma'], [u'asti', u'uttara', u'syAm', u'diSi', u'devat', u'AtmAs'], [u'asti', u'uttara', u'syAm', u'diSi', u'devata', u'at', u'mA'], [u'asti', u'uttara', u'syAm', u'diSi', u'devata', u'at', u'mAs'], [u'asti', u'uttara', u'syAm', u'diSi', u'devata', u'A', u'at', u'mA'], [u'asti', u'uttara', u'syAm', u'diSi', u'devata', u'A', u'at', u'mAs'], [u'asti', u'uttara', u'syAm', u'diSi', u'devata', u'AtmA'], [u'asti', u'uttara', u'syAm', u'diSi', u'devata', u'Atma'], [u'asti', u'uttara', u'syAm', u'diSi', u'devata', u'AtmAs'], [u'asti', u'uttara', u'syAm', u'diSi', u'devatA', u'at', u'mA'], [u'asti', u'uttara', u'syAm', u'diSi', u'devatA', u'at', u'mAs'], [u'asti', u'uttara', u'syAm', u'diSi', u'devatA', u'A', u'at', u'mA'], [u'asti', u'uttara', u'syAm', u'diSi', u'devatA', u'A', u'at', u'mAs'], [u'asti', u'uttara', u'syAm', u'diSi', u'devatA', u'AtmA'], [u'asti', u'uttara', u'syAm', u'diSi', u'devatA', u'Atma'], [u'asti', u'uttara', u'syAm', u'diSi', u'devatA', u'AtmAs'], [u'asti', u'uttaras', u'yAm', u'diSi', u'de', u'vatA', u'at', u'mA'], [u'asti', u'uttaras', u'yAm', u'diSi', u'de', u'vatA', u'at', u'mAs'], [u'asti', u'uttaras', u'yAm', u'diSi', u'de', u'vatA', u'A', u'at', u'mA'], [u'asti', u'uttaras', u'yAm', u'diSi', u'de', u'vatA', u'A', u'at', u'mAs'], [u'asti', u'uttaras', u'yAm', u'diSi', u'de', u'vatA', u'AtmA'], [u'asti', u'uttaras', u'yAm', u'diSi', u'de', u'vatA', u'Atma'], [u'asti', u'uttaras', u'yAm', u'diSi', u'de', u'vatA', u'AtmAs'], [u'asti', u'uttaras', u'yAm', u'diSi', u'de', u'vatAt', u'mA'], [u'asti', u'uttaras', u'yAm', u'diSi', u'de', u'vatAt', u'mAs'], [u'asti', u'uttaras', u'yAm', u'diSi', u'de', u'avat', u'A', u'at', u'mA'], [u'asti', u'uttaras', u'yAm', u'diSi', u'de', u'avat', u'A', u'at', u'mAs'], [u'asti', u'uttaras', u'yAm', u'diSi', u'de', u'avat', u'AtmA'], [u'asti', u'uttaras', u'yAm', u'diSi', u'de', u'avat', u'Atma'], [u'asti', u'uttaras', u'yAm', u'diSi', u'de', u'avat', u'AtmAs'], [u'asti', u'uttaras', u'yAm', u'diSi', u'de', u'avata', u'at', u'mA'], [u'asti', u'uttaras', u'yAm', u'diSi', u'de', u'avata', u'at', u'mAs'], [u'asti', u'uttaras', u'yAm', u'diSi', u'de', u'avata', u'A', u'at', u'mA'], [u'asti', u'uttaras', u'yAm', u'diSi', u'de', u'avata', u'A', u'at', u'mAs'], [u'asti', u'uttaras', u'yAm', u'diSi', u'de', u'avata', u'AtmA'], [u'asti', u'uttaras', u'yAm', u'diSi', u'de', u'avata', u'Atma'], [u'asti', u'uttaras', u'yAm', u'diSi', u'de', u'avata', u'AtmAs'], [u'asti', u'uttaras', u'yAm', u'diSi', u'de', u'avatA', u'at', u'mA'], [u'asti', u'uttaras', u'yAm', u'diSi', u'de', u'avatA', u'at', u'mAs'], [u'asti', u'uttaras', u'yAm', u'diSi', u'de', u'avatA', u'A', u'at', u'mA'], [u'asti', u'uttaras', u'yAm', u'diSi', u'de', u'avatA', u'A', u'at', u'mAs'], [u'asti', u'uttaras', u'yAm', u'diSi', u'de', u'avatA', u'AtmA'], [u'asti', u'uttaras', u'yAm', u'diSi', u'de', u'avatA', u'Atma'], [u'asti', u'uttaras', u'yAm', u'diSi', u'de', u'avatA', u'AtmAs'], [u'asti', u'uttaras', u'yAm', u'diSi', u'devat', u'A', u'at', u'mA'], [u'asti', u'uttaras', u'yAm', u'diSi', u'devat', u'A', u'at', u'mAs'], [u'asti', u'uttaras', u'yAm', u'diSi', u'devat', u'AtmA'], [u'asti', u'uttaras', u'yAm', u'diSi', u'devat', u'Atma'], [u'asti', u'uttaras', u'yAm', u'diSi', u'devat', u'AtmAs'], [u'asti', u'uttaras', u'yAm', u'diSi', u'devata', u'at', u'mA'], [u'asti', u'uttaras', u'yAm', u'diSi', u'devata', u'at', u'mAs'], [u'asti', u'uttaras', u'yAm', u'diSi', u'devata', u'A', u'at', u'mA'], [u'asti', u'uttaras', u'yAm', u'diSi', u'devata', u'A', u'at', u'mAs'], [u'asti', u'uttaras', u'yAm', u'diSi', u'devata', u'AtmA'], [u'asti', u'uttaras', u'yAm', u'diSi', u'devata', u'Atma'], [u'asti', u'uttaras', u'yAm', u'diSi', u'devata', u'AtmAs'], [u'asti', u'uttaras', u'yAm', u'diSi', u'devatA', u'at', u'mA'], [u'asti', u'uttaras', u'yAm', u'diSi', u'devatA', u'at', u'mAs'], [u'asti', u'uttaras', u'yAm', u'diSi', u'devatA', u'A', u'at', u'mA'], [u'asti', u'uttaras', u'yAm', u'diSi', u'devatA', u'A', u'at', u'mAs'], [u'asti', u'uttaras', u'yAm', u'diSi', u'devatA', u'AtmA'], [u'asti', u'uttaras', u'yAm', u'diSi', u'devatA', u'Atma'], [u'asti', u'uttaras', u'yAm', u'diSi', u'devatA', u'AtmAs'], [u'asti', u'uttarasyAm', u'diSi', u'de', u'vatA', u'at', u'mA'], [u'asti', u'uttarasyAm', u'diSi', u'de', u'vatA', u'at', u'mAs'], [u'asti', u'uttarasyAm', u'diSi', u'de', u'vatA', u'A', u'at', u'mA'], [u'asti', u'uttarasyAm', u'diSi', u'de', u'vatA', u'A', u'at', u'mAs'], [u'asti', u'uttarasyAm', u'diSi', u'de', u'vatA', u'AtmA'], [u'asti', u'uttarasyAm', u'diSi', u'de', u'vatA', u'Atma'], [u'asti', u'uttarasyAm', u'diSi', u'de', u'vatA', u'AtmAs'], [u'asti', u'uttarasyAm', u'diSi', u'de', u'vatAt', u'mA'], [u'asti', u'uttarasyAm', u'diSi', u'de', u'vatAt', u'mAs'], [u'asti', u'uttarasyAm', u'diSi', u'de', u'avat', u'A', u'at', u'mA'], [u'asti', u'uttarasyAm', u'diSi', u'de', u'avat', u'A', u'at', u'mAs'], [u'asti', u'uttarasyAm', u'diSi', u'de', u'avat', u'AtmA'], [u'asti', u'uttarasyAm', u'diSi', u'de', u'avat', u'Atma'], [u'asti', u'uttarasyAm', u'diSi', u'de', u'avat', u'AtmAs'], [u'asti', u'uttarasyAm', u'diSi', u'de', u'avata', u'at', u'mA'], [u'asti', u'uttarasyAm', u'diSi', u'de', u'avata', u'at', u'mAs'], [u'asti', u'uttarasyAm', u'diSi', u'de', u'avata', u'A', u'at', u'mA'], [u'asti', u'uttarasyAm', u'diSi', u'de', u'avata', u'A', u'at', u'mAs'], [u'asti', u'uttarasyAm', u'diSi', u'de', u'avata', u'AtmA'], [u'asti', u'uttarasyAm', u'diSi', u'de', u'avata', u'Atma'], [u'asti', u'uttarasyAm', u'diSi', u'de', u'avata', u'AtmAs'], [u'asti', u'uttarasyAm', u'diSi', u'de', u'avatA', u'at', u'mA'], [u'asti', u'uttarasyAm', u'diSi', u'de', u'avatA', u'at', u'mAs'], [u'asti', u'uttarasyAm', u'diSi', u'de', u'avatA', u'A', u'at', u'mA'], [u'asti', u'uttarasyAm', u'diSi', u'de', u'avatA', u'A', u'at', u'mAs'], [u'asti', u'uttarasyAm', u'diSi', u'de', u'avatA', u'AtmA'], [u'asti', u'uttarasyAm', u'diSi', u'de', u'avatA', u'Atma'], [u'asti', u'uttarasyAm', u'diSi', u'de', u'avatA', u'AtmAs'], [u'asti', u'uttarasyAm', u'diSi', u'devat', u'A', u'at', u'mA'], [u'asti', u'uttarasyAm', u'diSi', u'devat', u'A', u'at', u'mAs'], [u'asti', u'uttarasyAm', u'diSi', u'devat', u'AtmA'], [u'asti', u'uttarasyAm', u'diSi', u'devat', u'Atma'], [u'asti', u'uttarasyAm', u'diSi', u'devat', u'AtmAs'], [u'asti', u'uttarasyAm', u'diSi', u'devata', u'at', u'mA'], [u'asti', u'uttarasyAm', u'diSi', u'devata', u'at', u'mAs'], [u'asti', u'uttarasyAm', u'diSi', u'devata', u'A', u'at', u'mA'], [u'asti', u'uttarasyAm', u'diSi', u'devata', u'A', u'at', u'mAs'], [u'asti', u'uttarasyAm', u'diSi', u'devata', u'AtmA'], [u'asti', u'uttarasyAm', u'diSi', u'devata', u'Atma'], [u'asti', u'uttarasyAm', u'diSi', u'devata', u'AtmAs'], [u'asti', u'uttarasyAm', u'diSi', u'devatA', u'at', u'mA'], [u'asti', u'uttarasyAm', u'diSi', u'devatA', u'at', u'mAs'], [u'asti', u'uttarasyAm', u'diSi', u'devatA', u'A', u'at', u'mA'], [u'asti', u'uttarasyAm', u'diSi', u'devatA', u'A', u'at', u'mAs'], [u'asti', u'uttarasyAm', u'diSi', u'devatA', u'AtmA'], [u'asti', u'uttarasyAm', u'diSi', u'devatA', u'Atma'], [u'asti', u'uttarasyAm', u'diSi', u'devatA', u'AtmAs']]
